apps/explorer/lib/explorer/microservice_interfaces/multichain_search.ex

```diff
@@ -0,0 +1,191 @@
+defmodule Explorer.MicroserviceInterfaces.MultichainSearch do
+  @moduledoc """
+  Module to interact with Multichain search microservice
+  """
+
+  alias Ecto.Association.NotLoaded
+  alias Explorer.Chain.Cache.NetVersion
+  alias Explorer.Chain.Hash
+  alias Explorer.Utility.Microservice
+  alias HTTPoison.Response
+
+  require Logger
+
+  @post_timeout :timer.minutes(5)
+  @request_error_msg "Error while sending request to Multichain Search Service"
+
+  @doc """
+    Performs a batch import of addresses, blocks, and transactions to the Multichain Search microservice.
+
+    ## Parameters
+    - `params`: A map containing:
+      - `addresses`: List of address structs.
+      - `blocks`: List of block structs.
+      - `transactions`: List of transaction structs.
+
+    ## Returns
+    - `{:ok, :service_disabled}`: If the integration with Multichain Search Service is disabled.
+    - `{:ok, result}`: If the import was successful.
+    - `{:error, reason}`: If an error occurred.
+  """
+  @spec batch_import(%{
+          addresses: list(),
+          blocks: list(),
+          transactions: list()
+        }) :: {:error, :disabled | <<_::416>> | Jason.DecodeError.t()} | {:ok, any()}
```

Comment:
```suggestion
  @spec batch_import(%{
          addresses: [Address.t()],
          blocks: [Block.t()],
          transactions: [Transaction.t()]
        }) :: {:error, :disabled | String.t() | Jason.DecodeError.t()} | {:ok, any()}
```

Also, need to replace `alias Explorer.Chain.Hash` with `alias Explorer.Chain.{Address, Block, Hash, Transaction}`